src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,31 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    Animal.alive.push(this);
+    this.name = name;
+    this.health = 100;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = !this.hiden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden) {
+      prey.health -= 50;
+      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Comment:
It would be better to put it inside the check `if (prey.health <= 0)`. In this case you will use filter not every time, but only when prey has no health